Source/WebInspectorUI/UserInterface/Views/ResourceTreeElement.js

```diff
@@ -84,7 +84,18 @@ WI.ResourceTreeElement = class ResourceTreeElement extends WI.SourceCodeTreeElem
         if (comparisonResult !== 0)
             return comparisonResult;
 
-        // Compare by title when the subtitles are the same.
+        // Compare by title without file extensions when the subtitles are the same.
+        function filenameWithoutExtension(resourceTitle) {
+            return resourceTitle.substring(0, resourceTitle.lastIndexOf("."));
```

Comment:
i actually wonder if maybe we should be sorting based on the _first_ `"."`

for example, id imagine `test.min.js` to be before `test-a.js`

if so, we also could maybe be a bit more efficient in how we compare our strings by using `split(".", 1)`
```
function splitFilename(resourceTitle) {
    let [name, extensions] = resourceTitle.split(".", 1);
    return {name, extensions: extensions || ""};
}
let splitA = splitFilename(a.mainTitle);
let splitB = splitFilename(b.mainTitle);
return splitA.name.extendedLocaleCompare(splitB.name) || splitA.extensions.extendedLocaleCompare(splitB.extensions);
```

not a big deal tho if you find that less clear to read

Reply:
This is a great idea! I implemented this in: https://github.com/WebKit/WebKit/pull/18350